Src/HALAL/Services/Communication/SPI/SPI.cpp

```diff
@@ -24,9 +24,14 @@ uint8_t SPI::inscribe(SPI::Peripheral& spi){
 
 	SPI::Instance* spi_instance = SPI::available_spi[spi];
 
+	spi_instance->SCK->alternative_function = AF5;
+	spi_instance->MOSI->alternative_function = AF5;
+	spi_instance->MISO->alternative_function = AF5;
```

Comment:
what is this AF5?

Reply:
it doesn t do anything yet, as the alternative function pin declaration for spi is hard coded in the stm32_h7xx_hal_msp.c and is not handled by the ST-LIB (most of the modules of the ST-LIB do this (for now)), which makes the alternate function argument of the Pin unused. 
_(in reality the HAL methods on the stm32_h7xx_hal_msp.c are called by the ST-LIB, they just don t use the Pin structure inside them as you cannot pass them such parameters)_
This is a placeholder for a future update where we stop using the HAL initialization functions (if we decide to do that); and a way to synchronize the structure configuration to the actual hardcoded hardware configuration used by the peripheral. These lines being there or not should not change the behavior of the module (i tested it and it works on my computer), so in reality they could be removed, but i find weird having the Pin structure display AF0 when in reality the Pin is running AF5.